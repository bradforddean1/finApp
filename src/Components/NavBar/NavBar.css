.NavBar {
	display: flex;
}

.NavBar a {
	flex: 1;
	text-align: center;
	padding: 12.5px;
}
