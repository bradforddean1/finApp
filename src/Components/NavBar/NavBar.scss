@use "../../scss/variables" as v;

.NavBar {
	display: flex;
}

.NavBar a {
	flex: 1;
	text-align: center;
	padding: 12.5px;
}

@media screen and (min-width: v.$first-break-point) {
	.NavBar {
		height: 100%;
		flex-direction: column;
		align-items: start;
	}

	.NavBar a {
		flex-grow: 0;
	}
}

.NavBar figcaption {
	font-family: "Hind", sans-serif;
	font-style: normal;
	font-weight: 500;
	font-size: 0.44rem;
	line-height: 0.61rem;
}
