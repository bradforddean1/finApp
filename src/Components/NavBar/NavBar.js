import React from "react";
import { Link } from "react-router-dom";
import "./NavBar.css";
import Search from "../Common/icons/SearchIcon/SearchIcon";
import folder from "../../assets/folder.svg";
import exit from "../../assets/exit.svg";

/**
 * App naviagation bar
 * @component
 */
function NavBar(props) {
	return (
		<div className="NavBar bg-primary">
			<Link to="/search">
				<figure>
					<Search color={"#fff"} />
					<figcaption>Search</figcaption>
				</figure>
			</Link>
			<Link to="/portfolio">
				<figure>
					<img src={folder} alt="folder" />
					<figcaption>Portfolio</figcaption>
				</figure>
			</Link>
			<Link to="/logout">
				<figure>
					<img src={exit} alt="exit" />
					<figcaption>Logout</figcaption>
				</figure>
			</Link>
		</div>
	);
}

export default NavBar;
