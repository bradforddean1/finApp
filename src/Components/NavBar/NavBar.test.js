import React from "react";
import ReactDOM from "react-dom";
import NavBar from "./NavBar";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter } from "react-router-dom";

describe("Nav Component", () => {
	it("Renders", () => {
		const div = document.createElement("div");
		ReactDOM.render(
			<BrowserRouter>
				<NavBar />
			</BrowserRouter>,
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	});

	it("Renders the Default State", () => {
		const wrapper = shallow(<NavBar />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
