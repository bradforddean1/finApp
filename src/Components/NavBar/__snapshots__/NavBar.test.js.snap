// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component Renders the Default State 1`] = `
<div
  className="NavBar bg-primary"
>
  <Link
    to="/search"
  >
    <figure>
      <SearchIcon
        color="#fff"
      />
      <figcaption>
        Search
      </figcaption>
    </figure>
  </Link>
  <Link
    to="/portfolio"
  >
    <figure>
      <img
        alt="folder"
        src="folder.svg"
      />
      <figcaption>
        Portfolio
      </figcaption>
    </figure>
  </Link>
  <Link
    to="/logout"
  >
    <figure>
      <img
        alt="exit"
        src="exit.svg"
      />
      <figcaption>
        Logout
      </figcaption>
    </figure>
  </Link>
</div>
`;
